### ISSUE TITLE
[data grid] Unable to move out from a cell with a Link (React-Router) using arrow keys

### ISSUE THREAD

**Issue by sagar1111212121** [mui/mui-x#14815]
### The problem in depth

We are trying to make our site AA accessibility compliant. What we have observed in the DataGrid is, arrow keys are working all well until we have custom columns using cellRender where we have used react-router's Link. 

We are able to focus it using arrow key but once focus is set in that cell, it is not coming out using arrow keys.

We have checked on the MUI Accessibility section 'https://v6.mui.com/x/react-data-grid/accessibility/#system-GridAriaV7.tsx'. there also we see similar issue where, once focus is on email column, user is unable to move out to next column using arrow key.

![Image](https://github.com/user-attachments/assets/e28e90df-0bbe-4651-b3ac-e4da5e4d6f81)

```'javascript
 const fiDataGridColData: GridColDef[] = [
    {
      field: NAME_FIELD,
      headerName: NAME,
      disableColumnMenu: true,
      flex: 2,
      renderCell: (params: any) => {
        const fiNameWithoutSpace = `${params?.row?.name.replaceAll(/\s/g, '')}FI_Link`;
        return (
          <Tooltip description={params.row.name} placement='top'>
            <Link
              tabIndex={params.tabIndex}
              to={`/financialinstitution/${params.row.fiId}`}>
              {params.row.name}
            </Link>
          </Tooltip>
        );
      },
    },
    {
      field: CLIENT_ID_FIELD,
      headerName: CLIENT_ID,
      disableColumnMenu: true,
      flex: 1,
    },
  ];

   <DataGridPro
        rows={fiDataGridRowData}
        columns={fiDataGridColData}
        ....
   />
```

### Your environment

<details>
  <summary>`npx @mui/envinfo`</summary>

```
  Chrome Browser
  Output from `npx @mui/envinfo` goes here.
```
</details>

**Search keywords**: Datagrid accessibility
**Order ID**: 81479

**Comment by romgrk** [comment]
Could you please share your reproduction case in a runnable sandbox? You can fork any of the docs demo using these buttons: https://mui.com/x/introduction/support/#bug-reproductions

**Comment by sagar1111212121** [comment]
https://codesandbox.io/p/sandbox/4kndfg?file=%2Fsrc%2FDemo.tsx%3A23%2C18

This is MUI Datagrid demo only. So what we have observed is, Arrow keys on link and custom column having button or icon button works just fine in normal case but if we have screen reader like 'NVDA' running, arrow key stuck at such link/custom column having buttons and it does not shift left or right on arrow key press. 

You can try same on above by running NVDA and without NVDA.

**Comment by romgrk** [comment]
Could you put your code inside that codesandbox? Or is the problem reproducible with the demo as it is?

**Comment by sagar1111212121** [comment]
Yes, you can reproduce issue with this demo itself. Keep NVDA screen reader running and then try to use keyboard on the email column. It doesnt move out of email column

https://codesandbox.io/p/sandbox/4kndfg?file=%2Fsrc%2FDemo.tsx%3A23%2C18

**Comment by romgrk** [comment]
This problem is present in other grids and is an NVDA issue/bug:
 - https://github.com/nvaccess/nvda/issues/8395
 - https://github.com/nvaccess/nvda/issues/11413
 - https://github.com/nvaccess/nvda/issues/13476

I'm not sure there's anything we can do from here, it would be best to comment on the upstream issues. You can link to this issue to highlight the importance of fixing the issue in NVDA.
